### PR TITLE
Async tests

### PR DESCRIPTION
**Pull request by tmashuang**
Nuked Tests with async

### CODE REVIEW

File: test/unit/account-link-test.js [hunk 1]
```diff
@@ -2,14 +2,15 @@ var assert = require('assert')
 var linkGen = require('../../ui/lib/account-link')
 
 describe('account-link', function () {
-  it('adds ropsten prefix to ropsten test network', function () {
-    var result = linkGen('account', '3')
+  it('adds ropsten prefix to ropsten test network', async function (done) {
+    var result = await linkGen('account', '3')
```
Comment (flyswatter) [inline test/unit/account-link-test.js]: `async function (done) {` ftw!!!!

File: test/unit/account-link-test.js [hunk 2]
```diff
@@ -2,14 +2,15 @@ var assert = require('assert')
...
     assert.notEqual(result.indexOf('ropsten'), -1, 'ropsten included')
     assert.notEqual(result.indexOf('account'), -1, 'account included')
+    done()
   })
 
-  it('adds kovan prefix to kovan test network', function () {
-    var result = linkGen('account', '42')
+  it('adds kovan prefix to kovan test network', async function () {
```
Comment (flyswatter) [inline test/unit/account-link-test.js]: Shouldn't this test have a `done` param that gets called?

File: test/unit/actions/config_test.js [hunk 3]
```diff
@@ -21,20 +21,20 @@ describe('config view actions', function () {
   freeze(initialState)
 
   describe('SHOW_CONFIG_PAGE', function () {
-    it('should set appState.currentView.name to config', function () {
-      var result = reducers(initialState, actions.showConfigPage())
+    it('should set appState.currentView.name to config', async function () {
```
Comment (flyswatter) [inline test/unit/actions/config_test.js]: Again, shouldn't any `async` function use `done`?

**Review by danfinlay — COMMENTED** [review]
This is mostly fantastic and amazing, I just want to confirm that the `async` tests with no `done` param are in fact running.

File: test/unit/account-link-test.js (hunk 1 above)
Comment (kumavis) [inline test/unit/account-link-test.js]: should drop the done cb if this is an async fn

**Comment by kumavis** [comment]
@flyswatter i think you have it backwards -- if its an async fn its using the promise to tell that its resolved and should not use the done fn

File: test/unit/actions/config_test.js [hunk 4]
```diff
@@ -21,20 +21,20 @@ describe('config view actions', function () {
...
+      var result = await reducers(initialState, actions.showConfigPage())
```
Comment (kumavis) [inline test/unit/actions/config_test.js]: are you adding `await` to synchronous functions?

**Comment by kumavis** [comment]
@tmashuang i see lots of `await` being called on lots of previously synchronous functions - did you make the synchronous functions return promises?

**Comment by flyswatter** [comment]
Wait, mocha will accept promise-returning functions to end a test run?  That's news to me. I just thought the `done` param was the one way it supported async resolution.

Do `async` functions have implicit returns?  Quick test....  YES!

```
➜  node -v
v8.0.0
➜   node
> async function testThis() { console.log('ran') }
undefined
> testThis().then(() => {console.log('resolved')}).catch(err => console.error(err))
ran
Promise {
  <pending>,
  domain:
   Domain {
     domain: null,
     _events: { error: [Function: debugDomainError] },
     _eventsCount: 1,
     _maxListeners: undefined,
     members: [] } }
> resolved
```

File: test/unit/address-book-controller.js [hunk 5]
```diff
@@ -19,35 +19,39 @@ const mockKeyringController = {
 describe('address-book-controller', function () {
   var addressBookController
 
-  beforeEach(function () {
-    addressBookController = new AddressBookController({}, mockKeyringController)
+  beforeEach(async function () {
+    addressBookController = await new AddressBookController({}, mockKeyringController)
```
Comment (flyswatter) [inline test/unit/address-book-controller.js]: I see you using `await` with `new` in a few places.  Is this a thing? I don't think async construction is a thing.

Comment (kumavis) [inline test/unit/address-book-controller.js]: @tmashuang seems to have just put `await` in front of everything -- theres like 200 other examples in this PR

**Pull request closed without merging**